packages/sdk/test/evm/before-setup.ts

```diff
@@ -303,64 +324,177 @@ const generatePluginFunctions = (
+    MultichainRegistryCore__factory.bytecode,
   )
     .connect(signer)
-    .deploy()) as TWMultichainRegistryLogic;
-  const multichainRegistryLogic =
-    await multichainRegistryLogicDeployer.deployed();
+    .deploy()) as MultichainRegistryCore;
+  const multichainRegistryCore =
+    await multichainRegistryCoreDeployer.deployed();
 
-  const plugins: Plugin[] = generatePluginFunctions(
-    multichainRegistryLogic.address,
-    TWMultichainRegistryLogic__factory.abi,
+  const functionsCore: PluginFunction[] = generatePluginFunctions(
+    multichainRegistryCore.address,
+    MultichainRegistryCore__factory.abi,
+  );
+  const metadataCore: PluginMetadata = {
+    name: "MultichainRegistryCore",
+    metadataURI: "",
+    implementation: multichainRegistryCore.address,
+  };
+  plugins.push({
+    metadata: metadataCore,
+    functions: functionsCore,
+  });
+  pluginNames.push("MultichainRegistryCore");
+
+  // Add plugins to plugin-registry
+  await Promise.all(
+    plugins.map((plugin) => {
+      return extensionRegistry.addExtension(plugin);
+    }),
   );
 
-  const pluginMapDeployer = (await new ethers.ContractFactory(
-    PluginMap__factory.abi,
-    PluginMap__factory.bytecode,
-  )
-    .connect(signer)
-    .deploy(plugins)) as PluginMap;
-  const pluginMap = await pluginMapDeployer.deployed();
+  // Add util plugin names
+  pluginNames.push("PermissionsEnumerable");
+  pluginNames.push("ERC2771Context");
+
   const multichainRegistryRouterDeployer = (await new ethers.ContractFactory(
-    TWMultichainRegistryRouter__factory.abi,
-    TWMultichainRegistryRouter__factory.bytecode,
+    TWMultichainRegistry__factory.abi,
+    TWMultichainRegistry__factory.bytecode,
   )
     .connect(signer)
-    .deploy(pluginMap.address, [
-      trustedForwarderAddress,
-    ])) as TWMultichainRegistryRouter;
+    .deploy(extensionRegistry.address, pluginNames)) as TWMultichainRegistry;
   const multichainRegistryRouter =
     await multichainRegistryRouterDeployer.deployed();
 
   return multichainRegistryRouter.address;
 }
 
 // Setup marketplace-v3 for tests
-async function setupMarketplaceV3(): Promise<string> {
+async function setupMarketplaceV3(
+  trustedForwarderAddress: string,
+): Promise<string> {
+  const plugins: Plugin[] = [];
+  const pluginNames: string[] = [];
+
+  // PermissionsEnumerable plugin
+  const permissionsAddress = await deployContractAndUploadMetadata(
+    PermissionsEnumerable__factory.abi,
+    PermissionsEnumerable__factory.bytecode,
+    signer,
+  );
+  const functionsPermissions: PluginFunction[] = generatePluginFunctions(
+    permissionsAddress,
+    PermissionsEnumerable__factory.abi,
+  );
+  const metadataPermissions: PluginMetadata = {
+    name: "PermissionsEnumerable",
+    metadataURI: "",
+    implementation: permissionsAddress,
+  };
+  plugins.push({
+    metadata: metadataPermissions,
+    functions: functionsPermissions,
+  });
+  pluginNames.push("PermissionsEnumerable");
```

Comment:
let's make a helper function for constructing a full extension struct - metadata + functions + names- and reuse everywhere